v2/yarpcprotobuf/protoc-gen-yarpc-go/internal/generator/registry.go

```diff
@@ -0,0 +1,293 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package generator
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/gogo/protobuf/protoc-gen-gogo/descriptor"
+	plugin "github.com/gogo/protobuf/protoc-gen-gogo/plugin"
+)
+
+const (
+	_implPrefix = "_"
+	_client     = "Client"
+	_server     = "Server"
+	_stream     = "Stream"
+	_fx         = "Fx"
+	_yarpc      = "YARPC"
+	_procedures = "Procedures"
+)
+
+// registry is used to collect and register all
+// of the Protobuf types relevant to protoc-gen-yarpc-go.
+type registry struct {
+	files    map[string]*File
+	messages map[string]*Message
+	imports  Imports
+
+	/* Plugin parameters */
+	packages map[string]string
+}
+
+func newRegistry(req *plugin.CodeGeneratorRequest) (*registry, error) {
+	r := &registry{
+		files:    make(map[string]*File),
+		messages: make(map[string]*Message),
+		packages: make(map[string]string),
+		imports: Imports{
+			"context":                            "context",
+			"go.uber.org/fx":                     "fx",
+			"github.com/gogo/protobuf/proto":     "proto",
+			"go.uber.org/yarpc/v2":               "yarpc",
+			"go.uber.org/yarpc/v2/yarpcprotobuf": "yarpcprotobuf",
+		},
+	}
+	// Process command-line plugin parameters.
+	if req.Parameter != nil {
+		for _, p := range strings.Split(req.GetParameter(), ",") {
+			param := strings.SplitN(p, "=", 2)
+			if len(param) != 2 {
+				return nil, fmt.Errorf("package modifiers should include a single '='")
+			}
+			flag, value := param[0], param[1]
+			switch {
+			case strings.HasPrefix(flag, "M"):
+				r.packages[flag[1:]] = value
+			default:
```

Comment:
Should this just be an ignore, rather than an error? aka, unknown flags are no-ops for this plugin?
https://github.com/golang/protobuf#parameters indicates the other parameters that may be passed.